### PR TITLE
feat: Support plugins with scoped names

### PR DESCRIPTION
**Pull request by scottnonnenberg**
I was getting this error:

```
Error: Cannot find module 'eslint-plugin-@scottnonnenberg/thehelp'
    at Function.Module._resolveFilename (module.js:325:15)
    at Function.Module._load (module.js:276:25)
    at Module.require (module.js:353:17)
    at require (internal/module.js:12:17)
    at getPluginRule (node_modules/eslint-find-rules/src/lib/rule-finder.js:41:26)
    ...
```

### CODE REVIEW

File: test/lib/rule-finder.js [hunk 1]
```diff
@@ -77,57 +85,107 @@ describe('rule-finder', function() {
 
   it('specifiedFile (relative path) is passed to the constructor', function() {
     var ruleFinder = getRuleFinder(specifiedFileRelative)
-    assert.deepEqual(ruleFinder.getUnusedRules(), ['baz-rule', 'react/bar-rule', 'react/baz-rule', 'react/foo-rule'])
+    assert.deepEqual(ruleFinder.getUnusedRules(), [
+      'baz-rule',
+      'react/bar-rule',
+      'react/baz-rule',
+      'react/foo-rule',
+      'something/two-rule',
```
Comment (sarbbottam) [inline test/lib/rule-finder.js]: Do you not expect `something/two-rule` to be `@someone/something/two-rule`?

Comment (ta2edchimp) [inline test/lib/rule-finder.js]: As I understand the [ESLint docs](http://eslint.org/docs/developer-guide/working-with-plugins), the rules are to be referenced by their containing plugin's name (regardless of their possibly scoped package's name).

So for `eslint-plugin-<plugin-name>`'s rules, as well as `@scope/eslint-plugin-<plugin-name>`, its rules should be configured via `<plugin-name>/<rule-name>`.

Please correct me, if I'm wrong.

Comment (sarbbottam) [inline test/lib/rule-finder.js]: May not be a practical use case, curious to know, how does `eslint` handle, two or more scoped plugins, of the same name?

File: src/lib/rule-finder.js [hunk 2]
```diff
@@ -33,16 +33,41 @@ function _getCurrentRules(config) {
...
+  var scopedRegex = /(@[^/]+)\/(.+)/
+  var match = scopedRegex.exec(name)
+
+  if (match) {
+    return match[2]
+  }
+
+  return name
+}
+
```
Comment (sarbbottam) [inline src/lib/rule-finder.js]: `_pluginNameToModule` and `_pluginNameToRulePrefix` is almost same other than the return value, could you combine them in to one? 

Something like so ...
```js
function _normalizePluginName(name) {
  var scopedRegex = /(@[^/]+)\/(.+)/
  var match = scopedRegex.exec(name)

  if (match) {
    return {
      module: match[1] + '/' + 'eslint-plugin-' + match[2],
      prefix: match[2]
    }
  }

  return {
    module: 'eslint-plugin-' + name
    prefix: name
  }
}
```

File: src/lib/rule-finder.js [hunk 3]
```diff
@@ -33,16 +33,41 @@ function _getCurrentRules(config) {
...
 function _getPluginRules(config) {
   var pluginRules = []
   var plugins = config.plugins
   if (plugins) {
     plugins.forEach(function getPluginRule(plugin) {
-      var pluginConfig = require('eslint-plugin-' + plugin)
+      var module = _pluginNameToModule(plugin)
+      var prefix = _pluginNameToRulePrefix(plugin)
+      var pluginConfig = require(module)
```
Comment (sarbbottam) [inline src/lib/rule-finder.js]: ... and make corresponding updates

File: test/lib/rule-finder.js [hunk 4]
```diff
@@ -19,6 +19,14 @@ var getRuleFinder = proxyquire('../../src/lib/rule-finder', {
     '@noCallThru': true,
     '@global': true,
   },
+  '@someone/eslint-plugin-something': {
+    rules: {
+      'one-rule': true,
+      'two-rule': true,
```
Comment (sarbbottam) [inline test/lib/rule-finder.js]: I would stick to `foo`, `bar` rules for consistency.
**And even modified the identifier names from `react` to `plugin` and `something` to `scoped-plugin`.**

When we revisit the code later on, the names will be self explanatory.

**Comment by sarbbottam** [comment]
@scottnonnenberg Thanks for the PR!

File: test/lib/rule-finder.js (hunk 1 above)
Comment (scottnonnenberg) [inline test/lib/rule-finder.js]: @ta2edchimp I tested it in another project as I was working on this, and yes, for configuration rules take only the plugin name and not the scope name. Wasn't my first guess.

**Comment by scottnonnenberg** [comment]
That should take care of all the feedback - let me know if you've got anything else!

**Comment by sarbbottam** [comment]
👍 looks great! Thanks!

@ta2edchimp please have a you take a look and merge if you don't have any comments.

**Comment by sarbbottam** [comment]
Ohh, @scottnonnenberg please add yourself to the [contributors list](https://github.com/sarbbottam/eslint-find-rules/blob/master/.all-contributorsrc)

**Comment by ta2edchimp** [comment]
Really cool, thanks for the PR!

**Pull request merged**